cli/src/cli_util.rs

```diff
@@ -2714,10 +2716,10 @@ impl CliRunner {
     #[instrument(skip(self))]
     pub fn run(mut self) -> ExitCode {
         let mut default_config = crate::config::default_config();
-        if let Some(extra_configs) = self.extra_configs.take() {
+        for extra_config in self.extra_configs.drain(..) {
             default_config = config::Config::builder()
                 .add_source(default_config)
-                .add_source(extra_configs)
+                .add_source(extra_config)
```

Comment:
nit: no need to `.build()` for each config, the performance wouldn't matter though.

```rust
.fold(builder, |builder, config| builder.add_source(config))
```